src/views/RecIM/components/List/Listing/index.js

```diff
@@ -124,23 +128,30 @@ const ParticipantListing = ({ participant }) => {
             variant="rounded"
           ></Avatar>
         </ListItemAvatar>
-        <Link to={`/profile/${participant.username}`} className="gc360_link">
-          <ListItemText primary={participant.username} />
+        <Link to={`/profile/${participant.Username}`} className="gc360_link">
+          <ListItemText primary={name} />
         </Link>
       </Grid>
     </ListItem>
   );
 };
 
-const MatchListing = ({ match }) => {
+const MatchListing = ({ match, activityID }) => {
+  if (!match?.Team?.length) {
+    console.log('Error: MatchListing missing required info; this should be handled elsewhere');
+    return null;
+  }
+
   return (
     <ListItemButton
       component={Link}
-      to={`/recim/activity/${match.activityID}/match/${match.ID}`}
+      to={`/recim/activity/${activityID}/match/${match.ID}`}
       className="gc360_link"
     >
       <Grid container className={styles.listing}>
-        <Grid item>Team A vs Team B</Grid>
+        <Grid item>
+          {match.Team[0].Name} vs. {match.Team[1].Name}
+        </Grid>
```

Comment:
@silasw2, I think eventually we need to find a way to separate these team "columns" or somehow handle matches with more than 2 teams. For now this is okay.

Reply:
@amos-cha Yea, I am going to loop over them, will come with my next commit. Thanks for pointing it out!

Reply:
I actually think that for now, it might be best to leave it like this, and our UI design for the team page will need to change in the future to be able to handle multiple events. There just needs to be a bit more brainstorming around how it will work.